packages/host/app/components/operator-mode/add-field-modal.gts

```diff
@@ -123,46 +125,61 @@ export default class AddFieldModal extends Component<Signature> {
       relationshipType = this.cardinality === 'one' ? 'linksTo' : 'linksToMany';
     }
 
-    this.args.moduleSyntax.addField(
-      {
-        type: 'exportedName',
-        name: (
-          identifyCard(this.args.card)! as { module: string; name: string }
-        ).name,
-      },
-      this.fieldName,
-      this.chosenCatalogEntry.ref,
-      relationshipType,
-      new URL(this.chosenCatalogEntry.id),
-      this.loaderService.loader.reverseResolution(
-        makeResolvedURL(this.operatorModeStateService.state.codePath!).href,
-      ),
-      new URL(this.args.file.realmURL),
-    );
+    try {
+      this.args.moduleSyntax.addField(
+        {
+          type: 'exportedName',
+          name: (
+            identifyCard(this.args.card)! as { module: string; name: string }
+          ).name,
+        },
+        this.fieldName,
+        this.chosenCatalogEntry.ref,
+        relationshipType,
+        new URL(this.chosenCatalogEntry.id),
+        this.loaderService.loader.reverseResolution(
+          makeResolvedURL(this.operatorModeStateService.state.codePath!).href,
+        ),
+        new URL(this.args.file.realmURL),
+      );
+    } catch (error) {
+      let errorMessage = (error as Error).message;
+      if (errorMessage.includes('already exists')) {
```

Comment:
This is sensitive to the message thrown in `moduleSyntax` and it could break if someone changed the message. But we now have a test that makes sure the message is shown so it shouldn't be an issue